src/view/dataElements/identityMap/utils/namespacesUtils.js

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import fetchSandboxes from "../../xdmObject/helpers/fetchSandboxes";
+import fetchNamespaces from "./fetchNamespaces";
+
+const isNotECID = namespace => {
+  return namespace.code !== "ECID";
+};
+
+const getNamespaces = async (initInfo, sandbox) => {
+  const namespaces = await fetchNamespaces({
+    orgId: initInfo.company.orgId,
+    imsAccess: initInfo.tokens.imsAccess,
+    sandbox
+  });
+
+  return namespaces || [];
+};
+
+const getDefaultSandbox = sandboxes => {
+  return sandboxes.find(sandbox => sandbox.isDefault);
+};
+
+const getNamespacesForSandboxesPromises = async (initInfo, sandboxes) => {
+  if (sandboxes.size > 0) {
+    const promises = [];
+    sandboxes.forEach(sandbox => {
+      promises.push(getNamespaces(initInfo, sandbox));
+    });
+    return promises;
+  }
+
+  return fetchSandboxes({
+    orgId: initInfo.company.orgId,
+    imsAccess: initInfo.tokens.imsAccess
+  })
+    .then(result => {
+      return getDefaultSandbox(result.results);
+    })
+    .then(sandbox => getNamespaces(initInfo, sandbox.name));
```

Comment:
This should probably return an array of promises just like it does when sandboxes are defined on the extension settings.

Reply:
thank you, I have refactored this code a little bit